conan/tools/cmake/toolchain.py

```diff
@@ -425,34 +438,61 @@ def context(self):
         if prefer_config is not None and prefer_config.lower() in ("false", "0", "off"):
             find_package_prefer_config = "OFF"
 
-        os_ = self._conanfile.settings.get_safe("os")
-        android_prefix = "${CMAKE_CURRENT_LIST_DIR}" if os_ == "Android" else None
+        module_paths = []
 
+        # Read information from host context
         host_req = self._conanfile.dependencies.host.values()
-        cross_ios = os_ in ('iOS', "watchOS", "tvOS")
-
-        # Read the buildirs
-        build_paths = []
+        prefix_paths = []
+        lib_paths = []
+        framework_paths = []
+        include_paths = []
         for req in host_req:
             cppinfo = req.cpp_info.copy()
             cppinfo.aggregate_components()
-            build_paths.extend([os.path.join(req.package_folder,
-                                       p.replace('\\', '/').replace('$', '\\$').replace('"', '\\"'))
-                                for p in cppinfo.builddirs])
-
-        if self._toolchain.find_builddirs:
-            build_paths = " ".join(['"{}"'.format(b.replace('\\', '/')
+            module_paths.extend([os.path.join(req.package_folder, p) for p in cppinfo.builddirs if p != ""])
+            prefix_paths.extend([os.path.join(req.package_folder, p) for p in cppinfo.builddirs if p != ""])
+            lib_paths.extend([os.path.join(req.package_folder, p) for p in cppinfo.libdirs])
+            framework_paths.extend([os.path.join(req.package_folder, p) for p in cppinfo.frameworkdirs])
+            include_paths.extend([os.path.join(req.package_folder, p) for p in cppinfo.includedirs])
+
+        # Read information from build context
+        build_req = self._conanfile.dependencies.build.values()
+        bin_paths = []
+        for req in build_req:
```

Comment:
I am not sure we should be adding by default the paths to the "include", "libs" of ``build_requires``. Consider those are things like ``cmake`` with a transitive dependency of ``openssl`` and ``zlib``. If you do a find_library(openssl) in your code, almost never you will want to find the cmake->openssl build context library.

This is the reason why we don't create ``xxx-config.cmake`` by default for ``build_requires``. Recipes need to opt-in into that, because it is so easy that the ``zlib-config.cmake`` of build-context will conflict and potentially overwrite the one from the host context. I think exactly the same reasoning apply for findind libraries, includes, frameworks (and mostly everything except executables)

Reply:
It doesn't add `include` and `lib` paths of `build requires`, only `bin` and folders of CMake modules files (builddirs without root package folder).

Reply:
> And the executables and shared libs don't need to be added to the path, because they will already be in the PATH and LD_LIBRARY_PATH, because they are put there by the VirtualBuildEnv (which is a bit more general solution, as it works for every build system)

It is not sufficient for `find_*()` commands:
- `PATH ` has low precedence and might be descoped by `CMAKE_FIND_ROOT_PATH_MODE_PROGRAM ONLY`
- `LD_LIBRARY_PATH` doesn't matter for `find_*()` commands, it's a runtime variable.

Reply:
> It doesn't add include and lib paths of build requires, only bin and folders of CMake modules files (builddirs without root package folder).

Oh, yes you are right. I got a bit confused by the indents here in Github reviews with the truncated lines I lost scope.